code/datums/components/armor/synthetic_armor.dm

```diff
@@ -0,0 +1,60 @@
+// An edit of ablative armor, taken from Baystation12.
+/datum/component/armor/synthetic
+	full_block_message = "Your external plating absorbs the blow!"
+	partial_block_message = "Your external plating dulls the blow!"
+
+	/// Maximum armor values.
+	var/list/max_armor_values
+	/// How fast armor degrades with blocked damage, with armor value reduced by [coef * damage taken]
+	var/armor_degradation_coef = 0.8
+	/// Wearer feedback.
+	var/list/last_reported_damage
+
+/datum/component/armor/synthetic/Initialize(list/armor, armor_type, _armor_degradation_speed)
+	. = ..()
+	max_armor_values = armor_values.Copy()
+	if(_armor_degradation_speed)
```

Comment:
```suggestion
	if(!isnull(_armor_degradation_speed))
```
it might be worth using `!isnull()` here so that you could pass `0` and have armor that never degrades. who knows when someone might want to code something insanely OP for an event or something